### ISSUE TITLE
IR Parser should support block arguments and results. 

### ISSUE THREAD

**Issue by bgogul** [google-research/raksha#596]
Right now, there is no way to represent block arguments in the textual representation of the IR.  We should define the necessary syntax and update the parser.

This is not high priority yet.

**Comment by Cypher1** [comment]
I think both #617 and #592 are blocked on this at the moment.
Looking forward to further info, happy to take an action item / sub-task to help :)

**Comment by bgogul** [comment]
Are you completely blocked on #592? Can't we make progress on it with the understanding that the functionality will be limited?

Also, did you mean something other than #617?

**Comment by Cypher1** [comment]
Thanks Gogul, not completely blocked but only two of my test cases do not use inter-block references [as a place holder for block arguments and return values] (as this is core Raksha functionality imo).

I'll comment the tests that rely on this out with a TODO linking to this in the meantime, but in future I think we should avoid removing functionality when we don't have an alternative (especially when we have team's depending on us). Ideally there should have been some messaging about the removal too (I didn't realize this would be an outcome of the merged CLs).

**Comment by Cypher1** [comment]
About #617 Without block arguments + returns there are no references between blocks. Perhaps #617 is only working on in-block references to other operations, in which case I guess it's not blocked and we should file a separate issue for blocks.

**Comment by Cypher1** [comment]
Sorry, mentioned in chat but copying here for clarity: The parser now rejects references to operations from other blocks.

**Comment by harsha-mandadi** [comment]
> Thanks Gogul, not completely blocked but only two of my test cases do not use inter-block references [as a place holder for block arguments and return values] (as this is core Raksha functionality imo).
> 
> I'll split the tests that rely on this out into a separate PR with a TODO linking to this in the meantime, but in future I think we should avoid removing functionality when we don't have an alternative (especially when we have team's depending on us). Ideally there should have been some messaging about the removal too (I didn't realize this would be an outcome of the merged CLs).

J, looks like we already commented about the change in the PR. https://github.com/google-research/raksha/pull/592#discussion_r921377406. Let me know if that was not clear. :)

**Comment by Cypher1** [comment]
thanks Gogul,

It was the timeline that was unclear.
I tried to organise landing my change first but there was no response from
Harsha or you, and the breaking change was landed without a way forward
discussed.

Its sorted it out now, but it'd be nice to get existing PRs in before
breaking changes like this in future.
